crates/bevy_asset/src/server/mod.rs

```diff
@@ -1286,22 +1287,47 @@ pub enum AssetLoadError {
     },
 }
 
-/// An error that occurs when an [`AssetLoader`] is not registered for a given extension.
 #[derive(Error, Debug, Clone)]
+#[error("Failed to load asset '{path}' with asset loader '{loader_name}': {error}")]
+pub struct AssetLoaderError {
+    path: AssetPath<'static>,
+    loader_name: &'static str,
+    error: Arc<dyn std::error::Error + Send + Sync + 'static>,
+}
+
+impl PartialEq for AssetLoaderError {
+    #[inline]
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
We should call out in a doc string for this implementation that we're only comparing the `TypeId` of the underlying error.

This is somewhat surprising, so we should mention it to save people time when debugging.

Reply:
Probably want to use different phrasing than "partial" here:

> is partial (only through `TypeId`)

The "partial" in PartialEq refers to non-reflexiveness, not "partial comparison".